bugbug/github.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import logging
+from typing import Iterator, NewType, Tuple
+
+import requests
+from ratelimit import limits, sleep_and_retry
+
+from bugbug import db
+from bugbug.utils import get_secret
+
+logger = logging.getLogger(__name__)
+
+GITHUB_ISSUES_DB = "data/github_issues.json"
+db.register(
+    GITHUB_ISSUES_DB,
+    "https://community-tc.services.mozilla.com/api/index/v1/task/project.bugbug.data_github_issues.latest/artifacts/public/github_issues.json.zst",
+    1,
+)
+
+IssueDict = NewType("IssueDict", dict)
+
+GITHUB_TOKEN = get_secret("GITHUB_TOKEN")
+PER_PAGE = 100
+# Rate limit period in seconds
+RATE_LIMIT_PERIOD = 900
+
+
+def get_issues() -> Iterator[IssueDict]:
+    yield from db.read(GITHUB_ISSUES_DB)
+
+
+@sleep_and_retry
+@limits(calls=1200, period=RATE_LIMIT_PERIOD)
+def api_limit():
+    # Allow a limited number of requests to account for rate limiting
+    pass
+
+
+def fetch_events(events_url) -> str:
+    api_limit()
+    logger.info(f"Fetching {events_url}")
+    headers = {"Authorization": "token {}".format(GITHUB_TOKEN)}
+    response = requests.get(events_url, headers=headers)
+    response.raise_for_status()
+    events_raw = response.json()
+    return events_raw
+
+
+def fetch_issues(url, retrieve_events, params=None) -> Tuple[str, dict]:
+    try:
+        api_limit()
+        headers = {"Authorization": "token {}".format(GITHUB_TOKEN)}
+        response = requests.get(url, params=params, headers=headers)
+        response.raise_for_status()
+        data = response.json()
+
+        logger.info(f"Fetching {url}")
+
+        if retrieve_events:
+            for item in data:
+                try:
+                    events = fetch_events(item["events_url"])
+                    item.update({"events": events})
+                except Exception as e:
+                    print(e)
+                    raise
+
+        return data, response.links
+    except Exception as e:
+        print(e)
+        raise
```

Comment:
Nit: no need to print and reraise, you can remove the try-except and let the exception propagate